core/src/rpc.rs

```diff
@@ -1574,6 +1591,15 @@ fn verify_token_account_filter(
     }
 }
 
+fn check_confirmed_commitment(commitment: CommitmentConfig) -> Result<()> {
+    if commitment.is_processed() {
```

Comment:
I think here it might be worth it here to explicitly match on all the possible commitment levels, in case we add a weaker confirmation level in the future, the compiler will catch it.

Reply:
I mocked this up in `sdk/src/commitment_config.rs`, but I thought it made the code a whole lot less readable. However, I think the implementation now is a bit better, as new commitment levels will be default be `!is_at_least_confirmed` and have to be explicitly added to `is_confirmed` or `is_finalized`